tests/python/unittest/test_tir_schedule_tensorize_mfma.py

```diff
@@ -0,0 +1,322 @@
+
+
+def run_test(
+    k_inner,
+    in_dtype,
+    out_dtype,
+    b_transposed,
+    i_factors,
+    j_factors,
+    k_factors,
+    index_map_A,
+    index_map_B,
+    index_map_C,
+    ldmatrix_a_intrin,
+    ldmatrix_b_intrin,
+    mma_intrin,
+    mma_fill_intrin,
+    mma_store_intrin,
+):
+    sch = mfma_schedule(
+        te.create_prim_func(
+            matmul(M, N, K, in_dtype, out_dtype, b_transposed)),
+        k_inner,
+        in_dtype,
+        b_transposed,
+        i_factors,
+        j_factors,
+        k_factors,
+        index_map_A,
+        index_map_B,
+        index_map_C,
+        ldmatrix_a_intrin,
+        ldmatrix_b_intrin,
+        mma_intrin,
+        mma_fill_intrin,
+        mma_store_intrin,
+    )
+
+    f = tvm.build(sch.mod["main"], target="rocm -mcpu=gfx90a", name="dense")
+
+    dev = tvm.device("rocm", 0)
+    if in_dtype == "float32":
+        a_np = np.random.uniform(size=(M, K)).astype("float32")
+
+        if b_transposed:
+            b_np = np.random.uniform(size=(N, K)).astype("float32")
+            c_np = np.dot(a_np.astype("float32"), b_np.astype("float32").transpose()).astype(
+                out_dtype
+            )
+        else:
+            b_np = np.random.uniform(size=(K, N)).astype("float32")
+            c_np = np.dot(a_np.astype("float32"), b_np.astype(
+                "float32")).astype(out_dtype)
+    elif in_dtype == "float16":
+        a_np = np.random.uniform(size=(M, K)).astype("float16")
+
+        if b_transposed:
+            b_np = np.random.uniform(size=(N, K)).astype("float16")
+            c_np = np.dot(a_np.astype("float32"), b_np.astype("float32").transpose()).astype(
+                out_dtype
+            )
+        else:
+            b_np = np.random.uniform(size=(K, N)).astype("float16")
+            c_np = np.dot(a_np.astype("float32"), b_np.astype(
+                "float32")).astype(out_dtype)
+    else:
+        a_np = np.random.randint(-128, 128, (M, K)).astype("int8")
+
+        if b_transposed:
+            b_np = np.random.randint(-128, 128, (N, K)).astype("int8")
+            c_np = np.dot(a_np.astype("float32"), b_np.astype("float32").transpose()).astype(
+                "int32"
+            )
+        else:
+            b_np = np.random.randint(-128, 128, (K, N)).astype("int8")
+            c_np = np.dot(a_np.astype("float32"),
+                          b_np.astype("float32")).astype("int32")
+
+    a = tvm.nd.array(a_np, dev)
+    b = tvm.nd.array(b_np, dev)
+    c = tvm.nd.array(np.zeros((M, N), dtype=out_dtype), dev)
+
+    f(a, b, c)
+
+    if in_dtype != "float16":
+        # The numpy reference is computed with fp32 precision (otherwise too slow).
+        # So there is non-trivial accuracy difference if TVM result is computed with fp16 accumulation.
+        tvm.testing.assert_allclose(c.numpy(), c_np, rtol=1e-2, atol=1e-2)
+
+    return lambda: f.time_evaluator(f.entry_name, dev, number=500)(a, b, c)
+
+
+@tvm.testing.requires_rocm
```

Comment:
Does all rocm support matrix core?
If not, we can create a Feature like `requires_matrixcore`:

https://github.com/apache/tvm/blob/7767de9c39a724df6ca4f2bf028630d99f3ee746/python/tvm/testing/utils.py#L866C1-L871

So that we can skip test if matrixcore is not supported on CI.

Reply:
absolutely, I'm looking at how to get this information because we looks like don't provide hipcc wrapper like nvcc does. ( btw, I'm currently also trying to support hip source code gen, because llvm ir code generation is hard for debugging and integration, and if so, a hipcc wrapper.py will be made like nvcc.

Reply:
I think we can detect the matrix core support from rocm.compute_version, matrix core support starts from compute version 8.0.

https://github.com/LeiWang1999/tvm/blob/7f73cc47b6e86410511e1bbd5889957598d84735/python/tvm/contrib/rocm.py#L198-220

```python
>>> tvm.contrib.rocm.have_matrixcore()
True
```